File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/ad702387e4a493083d103e8e94dba895.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

type pair struct {
	first  int
	second int
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	n := nextInt(scanner)
	v := nextInt(scanner)
	m := nextInt(scanner)

	a := make([]int, n)
	b := make([]int, n)
	for i := 0; i < n; i++ {
		a[i] = nextInt(scanner)
	}
	for i := 0; i < n; i++ {
		b[i] = nextInt(scanner)
	}

	edges := make([][]int, n)
	for i := 0; i < m; i++ {
		x := nextInt(scanner) - 1
		y := nextInt(scanner) - 1
		edges[x] = append(edges[x], y)
		edges[y] = append(edges[y], x)
	}

	c := make([]int, n)
	visit := make([]int, n)
	cnt := 0
	s := make([]int, n)
	res := make([]pair, 0)

	var dfs func(int)
	dfs = func(x int) {
		visit[x] = 1
		c[x] = cnt
		for _, u := range edges[x] {
			if visit[u] == 0 {
				dfs(u)
			}
		}
	}

	for i := 0; i < n; i++ {
		if visit[i] == 0 {
			dfs(i)
			cnt++
		}
	}

	for i := 0; i < n; i++ {
		s[c[i]] += a[i]
		s[c[i]] -= b[i]
	}

	for i := 0; i < n; i++ {
		if a[i] == b[i] {
			continue
		}
		for j := i + 1; j < n; j++ {
			if c[i] != c[j] {
				continue
			}
			if a[j] == b[j] {
				continue
			}
			if a[i] < b[i] && a[j] < b[j] {
				continue
			}
			if a[i] > b[i] && a[j] > b[j] {
				continue
			}

			adjust(i, j, a, b, edges, res)
			break
		}
		if a[i] != b[i] {
			i--
		}
	}

	fmt.Println(len(res))
	for _, e := range res {
		fmt.Printf("%d %d %d\n", e.first+1, e.second+1, e.second)
	}
}

func adjust(x, y int, a, b []int, edges [][]int, res []pair) {
	if a[x] != b[x] {
		if a[y] != b[y] {
			if a[x] < b[x] {
				if a[y]+a[x] <= b[y] {
					liu(x, y, a, res)
					liu(y, x, a, res)
				} else {
					liu(x, y, b[y]-a[y], a, res)
					liu(y, x, b[y]-a[y], a, res)
					liu(x, y, a[x]-(b[y]-a[y]), a, res)
				}
			} else {
				if a[y]+b[x] <= b[y] {
					liu(x, y, a[y]-a[x], a, res)
					liu(y, x, a[y]-a[x], a, res)
					liu(x, y, b[x]-(a[y]-a[x]), a, res)
				} else {
					liu(x, y, b[y]-a[y], a, res)
					liu(y, x, b[y]-a[y], a, res)
				}
			}
		} else {
			if a[x] < b[x] {
				liu(x, y, a[x], a, res)
			} else {
				liu(x, y, b[x], a, res)
			}
		}
	} else {
		if a[y] != b[y] {
			if a[y] < b[y] {
				liu(y, x, b[y], a, res)
			} else {
				liu(y, x, a[y], a, res)
			}
		}
	}
}

func liu(x, y, flow int, a []int, res []pair) {
	a[x] -= flow
	a[y] += flow
	res = append(res, pair{first: x, second: y, second: flow})
}

func nextInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/C/ad702387e4a493083d103e8e94dba895.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_N 305

int n, v, m;
int a[MAX_N];
int b[MAX_N];
int edges[MAX_N][MAX_N];
int c[MAX_N];
int visit[MAX_N];
int cnt;
long long s[MAX_N];
int res[MAX_N][3];
int dist[MAX_N];

void dfs(int x) {
    visit[x] = 1;
    c[x] = cnt;
    for (int u = 0; u < n; u++) {
        if (edges[x][u] && !visit[u]) {
            dfs(u);
        }
    }
}

void bfs(int src) {
    memset(visit, 0, sizeof(visit));
    visit[src] = 1;
    dist[src] = 0;
    int queue[MAX_N];
    int front = 0, rear = 0;
    queue[rear++] = src;
    while (front < rear) {
        int x = queue[front++];
        for (int u = 0; u < n; u++) {
            if (edges[x][u] && !visit[u]) {
                dist[u] = dist[x] + 1;
                visit[u] = 1;
                queue[rear++] = u;
            }
        }
    }
}

void liu(int x, int y, int flow) {
    a[x] -= flow;
    a[y] += flow;
    res[cnt][0] = x + 1;
    res[cnt][1] = y + 1;
    res[cnt][2] = flow;
    cnt++;
}

void liu_helper(int i, int flow, int *routes) {
    if (i == routes[0]) {
        return;
    }
    int nxt = routes[i + 1];
    int now = routes[i];
    if (a[nxt] + flow <= v) {
        liu(now, nxt, flow);
        liu_helper(i + 1, flow, routes);
    } else {
        int original_a_now = a[now];
        int moved = v - a[nxt];
        liu(now, nxt, v - a[nxt]);
        liu_helper(i + 1, flow, routes);
        liu(now, nxt, flow - moved);
        int new_a_now = a[now];
    }
}

void adjust(int x, int y) {
    if (a[x] != b[x] && a[y] != b[y]) {
        if (a[x] < b[x]) {
            int temp = x;
            x = y;
            y = temp;
        }
        int flow = abs(a[x] - b[x]) < abs(a[y] - b[y]) ? abs(a[x] - b[x]) : abs(a[y] - b[y]);
        bfs(x);
        int routes[MAX_N];
        int route_len = 0;
        routes[route_len++] = y;
        while (y != x) {
            int step = dist[y];
            int nxt = -1;
            for (int ny = 0; ny < n; ny++) {
                if (edges[y][ny] && dist[ny] == step - 1) {
                    nxt = ny;
                    break;
                }
            }
            y = nxt;
            routes[route_len++] = y;
        }
        liu_helper(0, flow, routes);
    }
}

int main() {
    scanf("%d %d %d", &n, &v, &m);
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    for (int i = 0; i < n; i++) {
        scanf("%d", &b[i]);
    }
    memset(edges, 0, sizeof(edges));
    for (int i = 0; i < m; i++) {
        int x, y;
        scanf("%d %d", &x, &y);
        x--;
        y--;
        edges[x][y] = 1;
        edges[y][x] = 1;
    }
    memset(visit, 0, sizeof(visit));
    cnt = 0;
    for (int i = 0; i < n; i++) {
        if (!visit[i]) {
            dfs(i);
            cnt++;
        }
    }
    memset(s, 0, sizeof(s));
    for (int i = 0; i < n; i++) {
        s[c[i]] += a[i];
        s[c[i]] -= b[i];
    }
    for (int i = 0; i < n; i++) {
        if (s[i]) {
            printf("NO\n");
            return 0;
        }
    }
    int prev;
    int cur;
    cnt = 0;
    for (int i = 0; i < n; i++) {
        if (a[i] == b[i]) {
            continue;
        }
        for (int j = i + 1; j < n; j++) {
            if (c[i] != c[j]) {
                continue;
            }
            if (a[j] == b[j]) {
                continue;
            }
            if ((a[i] < b[i] && a[j] < b[j]) || (a[i] > b[i] && a[j] > b[j])) {
                continue;
            }
            adjust(i, j);
            break;
        }
        if (a[i] != b[i]) {
            i--;
        }
    }
    printf("%d\n", cnt);
    for (int i = 0; i < cnt; i++) {
        printf("%d %d %d\n", res[i][0], res[i][1], res[i][2]);
    }
    return 0;
}
